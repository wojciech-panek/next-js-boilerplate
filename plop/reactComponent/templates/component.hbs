import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';
import { compose } from 'ramda';

import withIntl from '../../shared/lib/withIntl'; // TODO fix relative import
import { Container } from './{{ camelCase name }}.styles';


export class {{ pascalCase name }} extends PureComponent {
  static propTypes = {
  };

  render() {
    return (
      <Container>
        {{ pascalCase name }} component
      </Container>
    );
  }
}

export default compose(
  withIntl,
)({{ pascalCase name }});

