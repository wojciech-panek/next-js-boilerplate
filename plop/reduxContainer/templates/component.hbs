import React, { PureComponent } from 'react';
import PropTypes from 'prop-types';


export class {{ pascalCase name }} extends PureComponent {
  static propTypes = {
  };

  render() {
    return (
      <div>
        {{ pascalCase name }} component
      </div>
    );
  }
}
