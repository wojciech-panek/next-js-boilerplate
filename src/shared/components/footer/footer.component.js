import React, { PureComponent } from 'react';
import { compose } from 'ramda';

import withIntl from '../../lib/withIntl';
import { Container } from './footer.styles';


export class Footer extends PureComponent {
  static propTypes = {
  };


  render() {
    return (
      <Container>
        Footer
      </Container>
    );
  }
}

export default compose(
  withIntl,
)(Footer);
