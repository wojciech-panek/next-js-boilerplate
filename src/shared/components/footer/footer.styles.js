import styled from 'styled-components';


export const Container = styled.footer`
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 20px 0;
`;

