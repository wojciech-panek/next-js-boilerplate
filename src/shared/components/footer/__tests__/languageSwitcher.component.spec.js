import React from 'react';
import { shallow } from 'enzyme';

import { Footer } from '../footer.component';


describe('Footer: Component', () => {
  const defaultProps = {
  };

  const component = (props) => (
    <Footer {...defaultProps} {...props} />
  );

  it('should render correctly', () => {
    const wrapper = shallow(component());
    global.expect(wrapper).toMatchSnapshot();
  });
});
