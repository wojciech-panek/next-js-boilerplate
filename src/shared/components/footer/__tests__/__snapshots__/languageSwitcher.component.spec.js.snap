// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Footer: Component should render correctly 1`] = `
<footerstyles__Container>
  Footer
</footerstyles__Container>
`;
